experimental/dds/tree2/src/feature-libraries/sequence-field/sequenceFieldToDelta.ts

```diff
@@ -74,10 +99,29 @@ function cellDeltaFromMark<TNodeChange>(
 				return mark.count;
 			}
 			case "Delete": {
-				return {
-					type: Delta.MarkType.Delete,
-					count: mark.count,
-				};
+				const detachedField = repairDataBuilder.handler({
```

Comment:
Each removed node needs to go in a separate detached field that corresponds to the ChangeAtomId for that node. So you'll need to move that code to be inside the loop and do `localId: mark.id + i` (modulo whatever the type system wants to do in order to make it type-check).